packages/wrangler/src/index.tsx

```diff
@@ -749,15 +754,79 @@ export async function main(argv: string[]): Promise<void> {
         // -- snip, end --
       }
 
+      // TODO: if worker_dev = false and no routes, then error (only for dev)
+
+      /**
+       * Given something that resembles a URL,
+       * try to extract a host from it
+       */
+      function getHost(urlLike: string): string | undefined {
+        const regex =
+          /^((http|https):\/\/)?(?<host>((\w|-)*\.)+((\w|-)*))(\/\w*)*((\/\*)|(\/))?$/imu;
+        return regex.exec(urlLike)?.groups?.host;
+      }
```

Comment:
You can just do

```suggestion
      function getHost(urlLike: string): string | undefined {
        const url = new URL(urlLike);
        return url.host;
      }
```

Or am I missing something?

Reply:
URL is a stricter, notably it throws if you don't include a protocol

Reply:
I admit I'm tempted to add a protocol if it doesn't exist and use the constructor to make this code simpler, hate having tio maintain regexes.